### PR TITLE
SONIC-1167 Add custom icon to BpkButton-Compose

### PR DESCRIPTION
**Pull request by mineschan**
Add drawable icon support to Compose version of BpkButton 

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

**Comment by mineschan** [comment]
Record snapshots

File: app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt [hunk 1]
```diff
@@ -44,6 +45,7 @@ class BpkButtonTest(flavour: Flavor) : BpkSnapshotTest() {
   private val type: BpkButtonType = flavour.first
   private val size: BpkButtonSize = flavour.second
   private val icon = BpkIcon.LongArrowRight
+  private val iconDrawableRes = R.drawable.button_sample_icon
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt:48]: can you use a different sample icon please? this might cause copyright issues.

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButton.kt [hunk 2]
```diff
@@ -133,6 +135,66 @@ fun BpkButton(
   }
 }
 
+@Composable
+fun BpkButton(
+  @DrawableRes iconResource: Int,
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButton.kt:140]: rather than fixing this to the `iconResource` (and therefore creating a dependency on the existing resource system) could this be an `icon` composable instead?

**Comment by marianeum** [comment]
Btw the record snapshots comment is no longer the appraoch to take - the `Record snapshots` commit is correct (we had to change it to a commit following some security concerns). Just make sure you wait for that run to succeed before making other commits.

File: app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt (hunk 1 above)
Comment (xo0ps) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt:49]: changed to backpack's icon

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButton.kt (hunk 2 above)
Comment (xo0ps) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButton.kt]: changed to use `Painter` instead

File: app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt [hunk 3]
```diff
@@ -96,7 +99,7 @@ class BpkButtonTest(flavour: Flavor) : BpkSnapshotTest() {
     // icon is bigger on large size, so we need to test this
 
     capture {
-      BpkButton("Button", icon, BpkButtonIconPosition.Start, type = type, size = size, onClick = {})
+      BpkButton("Button", icon, BpkButtonIconPosition.End, type = type, size = size, onClick = {})
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt:102]: why was this changed to end?

Comment (marianeum) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/button/BpkButtonTest.kt:102]: ah you're right! i got confused by the github formatting as it didn't include the function title

**Review by marianeum — APPROVED** [review]
